Movied/Movied/Scenes/Welcome/Cells/WelcomeCollectionCell.swift

```diff
@@ -0,0 +1,37 @@
+import UIKit
+
+protocol WelcomeCollectionCellDelegate: AnyObject {
+    func didTapButton(cell: WelcomeCollectionCell)
+}
+
+final class WelcomeCollectionCell: UICollectionViewCell {
+    @IBOutlet private weak var welcomeImageView: UIImageView!
+    @IBOutlet private weak var nextButton: UIButton!
+    
+    weak var delegate: WelcomeCollectionCellDelegate?
+    
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        self.delegate = nil
```

Comment:
Why it should be nil here?

Reply:
Because I use delegation to pass a CELL, so I set it nil to release memory and for safety

Reply:
No, when we declare the delegate as weak. It will not own any thing so will have no leak memory.